src/js/common/schemaform/actions.js

```diff
@@ -81,3 +114,182 @@ export function submitForm(formConfig, form) {
     );
   };
 }
+
+/**
+ * Transforms the data from an old version of a form to be used in the latest
+ *  version.
+ *
+ * @param  {Object}  savedData    The formData from the old version of the form.
+ * @param  {Ingeter} savedVersion The version of the form the corresponding
+ *                                 data was saved with.
+ * @param  {Array}   migrations   An array of functions which transform the
+ *                                 data saved to work with the current version.
+ * @return {Object}               The modified formData which should work with
+ *                                 the current version of the form.
+ */
+function getUpdatedFormData(savedData, savedVersion, migrations) {
+  // migrations is an array that looks like this:
+  // [
+  //   (savedData) => {
+  //     // Makes modifications to savedData to update it from version 0 -> version 1
+  //   },
+  //   (savedData) => {
+  //     // Makes modifications to update the data from version 1 -> version 2
+  //   },
+  //   ...
+  // ]
+  // The functions transform the data from version of their index to the next one up.
+  // This works because every time the version is bumped on the form, it's because
+  //  the saved data needs to be manipulated, so there will be no skipped versions.
+  if (!migrations || typeof migrations[savedVersion] !== 'function') {
+    return savedData;
+  }
+  return getUpdatedFormData(migrations[savedVersion](savedData), savedVersion + 1, migrations);
```

Comment:
I think in this case making this function recursive obscures what's happening, which is that you're iterating through the migrations and applying each one. I think `reduce` might be clearer.